webapp/channels/src/utils/desktop_app/auth.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import crypto from 'crypto';
+
+import {isDesktopApp} from 'utils/user_agent';
+
+export enum DesktopAuthStatus {
+    None,
+    Polling,
+    Expired,
+    Complete,
+}
+
+export const getExternalLoginURL = (url: string, search: string, token: string) => {
+    const params = new URLSearchParams(search);
+    if (isDesktopApp() && token) {
+        params.set('desktop_token', token);
+    }
+    const externalurl = `${url}${params.toString().length ? `?${params.toString()}` : ''}`;
```

Comment:
Could you split this line up a bit? I'm having a hard time parsing the nested template strings